internal/lagoonyml/required/lagoon.go

```diff
@@ -14,6 +14,11 @@ type LagoonRoute struct {
 	Ingresses map[string]Ingress
 }
 
+type LagoonCronjob struct {
+	Name    string
```

Comment:
This works, but the json unmarshaler will [search through permutations](https://go.dev/blog/json#decoding) of the field name to find the JSON field it's looking for. If you add a struct tag like `` `json:"name"` `` it shortcuts this behaviour and makes unmarshalling more predictable. See e.g. the `Environment` struct below.

Reply:
I was copying the style of [`LagoonRoute`](https://github.com/uselagoon/lagoon-linter/blob/main/internal/lagoonyml/required/lagoon.go#L13) which doesn't have tags. What's the difference between the two? Or should the route also have tags?

Reply:
Ah I see the confusion - LagoonRoute is different because it has an `UnmarshalJSON` [method](https://github.com/uselagoon/lagoon-linter/blob/main/internal/lagoonyml/required/lagoon.go#L18). That means it implements the `json.Unmarshaler` [interface](https://pkg.go.dev/encoding/json#Unmarshaler), so the default `json.Unmarshal` logic will be skipped and the custom `UnmarshalJSON` logic will be used instead. It's just the default unmarshal logic that uses the struct tags.

Reply:
TIL, thanks!